### PR TITLE
Rodriguez

### PR DESCRIPTION
**Pull request by robertorodriguez12**
finished league methods and added tests. All tests passed. Likely candidate for refactoring once we get to that point.

### CODE REVIEW

**Review by ShaunDaneJames — APPROVED** [review]
Wow Roberto, this look amazing!

File: lib/league_stats.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  def worst_offense
+    worst_offense = game_teams_by_team_id.min_by do |team_id, game_teams|
+      game_teams.sum{|game_team| game_team.goals} / game_teams.count.to_f
+    end.first
+
+    @teams.find{|team| team.team_id == worst_offense}.team_name
+  end
+
+  def find_all_away_teams
```
Comment (araspaas) [inline lib/league_stats.rb:28]: nice helper methods here

File: lib/league_stats.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  end
+
+  def best_offense
+    best_offense = game_teams_by_team_id.max_by do |team_id, game_teams|
+      game_teams.sum{|game_team| game_team.goals} / game_teams.count.to_f
+    end.first
+
+    @teams.find{|team| team.team_id == best_offense}.team_name
+  end
+
```
Comment (araspaas) [inline lib/league_stats.rb:19]: good job with making these two methods they look great.

**Review by araspaas — APPROVED** [review]
really like the helper methods they make it nice and short and very readable.

File: lib/game_team.rb [hunk 3]
```diff
@@ -9,7 +9,7 @@ def initialize(game_team_param)
     @result = game_team_param[:result]
     @settled_in = game_team_param[:settled_in]
     @head_coach = game_team_param[:head_coach]
-    @goals = game_team_param[:goals]
+    @goals = game_team_param[:goals].to_i
```
Comment (dcoleman21) [inline lib/game_team.rb:12]: Good thinking here to add the .to_i

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -6,9 +6,9 @@
 require_relative 'game_stats'
 
 class StatTracker
-  attr_reader :game_stats
+  attr_reader :game_stats,
+              :league_stats
```
Comment (dcoleman21) [inline lib/stat_tracker.rb:10]: Awesome add here with the league_stats

File: test/game_stats_test.rb [hunk 5]
```diff
@@ -12,7 +12,7 @@ class GameStatsTest < Minitest::Test
   def setup
     data = {
       games:      './data/games.csv',
-      teams:      './data/games.csv',
+      teams:      './data/teams.csv',
```
Comment (dcoleman21) [inline test/game_stats_test.rb:15]: good catch here with the teams.csv vs games.csv

**Review by dcoleman21 — COMMENTED** [review]
This looks good. We will run the spec_harness and double check everything as well.

**Pull request merged**